Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp

```diff
@@ -319,6 +285,17 @@ namespace AzFramework
             EBUS_EVENT_RESULT(currentEntity, AZ::ComponentApplicationBus, FindEntity, *entityIdIter);
             if (currentEntity)
             {
+                bool isPrefabSystemEnabled = false;
+                AzFramework::ApplicationRequests::Bus::BroadcastResult(
+                    isPrefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+                if (isPrefabSystemEnabled)
+                {
+                    SpawnedEntityTicketMapperInterface* spawnedEntityTicketMapperInterface =
+                        AZ::Interface<SpawnedEntityTicketMapperInterface>::Get();
+                    AZ_Assert(spawnedEntityTicketMapperInterface != nullptr, "SpawnedEntityTicketMapperInterface is not found.");
+                    spawnedEntityTicketMapperInterface->RemoveSpawnedEntity(currentEntity->GetId());
```

Comment:
Won't this cause the entity to not be deleted if it's not on a ticket?

Reply:
The 'DestroyGameEntityInternal' is not stopping at this point. This call will clear the entity from the ticket if a ticket is associated with it. The actual deletion of the entity still happens in the 'DestroyGameEntityInternal' call after this call.